Civi/Api4/Utils/ReflectionUtils.php

```diff
@@ -157,4 +157,49 @@ public static function getTraits($class) {
     return $traits;
   }
 
+  /**
+   * Get a list of standard properties which can be written+read by outside callers.
+   *
+   * @param string $class
+   */
+  public static function findStandardProperties($class): iterable {
+    try {
+      /** @var \ReflectionClass $clazz */
+      $clazz = new \ReflectionClass($class);
+
+      yield from [];
+      foreach ($clazz->getProperties(\ReflectionProperty::IS_PROTECTED | \ReflectionProperty::IS_PUBLIC) as $property) {
+        if (!$property->isStatic() && $property->getName()[0] !== '_') {
+          yield $property;
+        }
+      }
+    }
+    catch (\ReflectionException $e) {
+      throw new \RuntimeException(sprintf("Cannot inspect class %s.", $class));
+    }
+  }
+
+  /**
+   * Find any methods in this class which match the given prefix.
+   *
+   * @param string $class
+   * @param string $prefix
+   */
+  public static function findMethodHelpers($class, string $prefix): iterable {
+    try {
+      /** @var \ReflectionClass $clazz */
+      $clazz = new \ReflectionClass($class);
```

Comment:
if this parameter name out of pure hatred for IDEs with spell checkers?

Reply:
Muhahaha! The IDEs will explode!

"clazz" a Java-ism. In Java, you cannot have a variable named `class`, so they have a wide-spread convention to put the reflective-object into a variable named `clazz`. (See, eg, [stackexchange](https://stackoverflow.com/questions/2529974/why-do-java-programmers-like-to-name-a-variable-clazz/2529980)) And, by coincidence, Java was the first place where I learned to work with reflective objects...

In PHP, it is legal to have a variable `$class`, but you have another issue - it's very common to have `string` representations of a class-name. (IMHO, the `string` is more common than the `ReflectionClass`.) If both `string` and `ReflectionClass` are present, then you need two variable names. I fall back to `$clazz` as a familiar convention to unambiguously identify the `ReflectionClass` variable.

Of course, "familiar" is subjective and bikesheddy. IMHO it's more readable/writable to follow some kind of convention... if other conventions are more resonant here, that's fine...

* `string $class`/`ReflectionClass $clazz` (*used here*)
* `string $className`/`ReflectionClass $classObj` (*longer; maybe clearer to more people*)
* `string $class`/`ReflectionClass $refl` (*shorter; though `refl` is ambiguous about ReflClass vs ReflPropery vs ReflMethod*)

Reply:
Hmm OK - well I guess I can take that case back to my IDE & say 'so there'

Reply:
I would have gone with 2

string $className/ReflectionClass $classObj (longer; maybe clearer to more people)

Reply:
Actually I think I would push for 2 a bit harder - that feels like something we see a lot of across our codebase

![image](https://user-images.githubusercontent.com/336308/129307498-84c0811d-e425-410e-bdb5-96ef3bac83cc.png)

albeit not `$classObject` (or the nasty obj abbreviation )

I do find `$object` a lot and I'm pretty sure I can find things like `$resultObject` if I can think of the right string to look for

![image](https://user-images.githubusercontent.com/336308/129307596-ea5f69eb-929d-483e-931d-17e959a0b50c.png)

Reply:
![image](https://user-images.githubusercontent.com/336308/129307803-1a82f3ac-75f1-42e4-a4a7-7ef3e3924502.png)

Reply:
Did you search for `clazz`? On my local `dmaster`, `grep -r clazz|wc` finds ~160 matches. It pops up in Civi core+contrib as well as several upstream projects (jQuery-core, AngularJS-core, select2, monaco-editor, Quickbooks PHP API, etc). This isn't some new quirk sneaking in.

I'm a little frazzled here. This is an obscure local variable. And regardless of what we do with this particular variable -- we'll still have a muddy situation where `dmaster` and `universe` have a broad smattering of `class`, `clazz`, `reflectionClass`, `reflClass`, etc.

If we want to dig into the naming-convention (pros/cons of the various names -- are they pithy? wordy? evocative? intuitive? etc)... maybe that merits a distinct issue? And then follow-up to normalize the things within our control?

Reply:
@totten nah I can let it go

Reply:
but you are right - some of the issues we have are because we haven't discussed the merits of different priorities